### PR TITLE
chore: use dream2nix for a better npm experience

### PR DESCRIPTION
**Pull request by ulrikstrid**
## Problem

The current npm solution for nix doesn't work too well

## Solution

Use dream2nix that seems to work better with newer versions of npm.
This is a simplified/refactored version of #579

Also removed landmarks on `aarch64-darwin` since it's not compatible with the m1

As you can see here it unblocks m1: https://hydra.strid.ninja/jobset/deku/deku-m1

## Related

- #579 This is heavily inspired by the work in this PR so I'm adding co-author

### CODE REVIEW

File: nix/deku.nix [hunk 1]
```diff
@@ -14,15 +14,14 @@ in ocamlPackages.buildDunePackage rec {
       [ "dune-project" "sidechain.opam" "package.json" "package-lock.json" ];
   };
 
-  configurePhase = ''
-    export PATH=${npmPackages}/node_modules/.bin:$PATH
-    export NODE_PATH=${npmPackages}/node_modules
-
-    ln -s ${npmPackages}/node_modules ./node_modules
-  '';
+  #preBuildPhase = ''
```
Comment (d4hines) [inline nix/deku.nix:17]: Dead code?

File: nix/npm.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    packageOverrides = {
+      webpack-cli = {
+        remove-webpack-check = {
+          patches = [ ./patches/remove-webpack-check.patch ];
+        };
+      };
+    };
+
+    inject = { acorn-import-assertions."1.8.0" = [[ "acorn" "8.7.1" ]]; };
```
Comment (d4hines) [inline nix/npm.nix:18]: Curious, what is this?

File: nix/shell.nix [hunk 3]
```diff
@@ -1,9 +1,7 @@
-{ pkgs, deku, ligo, npmPackages }:
+{ pkgs, deku, ligo, nodejs ? pkgs.nodejs }:
 pkgs.mkShell {
   shellHook = ''
-    export NODE_PATH=${npmPackages}/node_modules
```
Comment (d4hines) [inline nix/shell.nix:4]: this is a slight loss because now we can't run nodejs locally with our node_modules pre-configured. However, I think it's reasonable, and it's easy to add it back if we need it.

File: nix/deku.nix (hunk 1 above)
Comment (ulrikstrid) [reply nix/deku.nix:17]: Yes, thank you

File: nix/npm.nix (hunk 2 above)
Comment (ulrikstrid) [reply nix/npm.nix:18]: I think it's a dependency of webpack, I tried removing it but then it complained, we should add a comment

File: nix/shell.nix (hunk 3 above)
Comment (ulrikstrid) [reply nix/shell.nix:4]: This is now wrapped into the webpack binary. We can easily revert this if you run it in another way

**Comment by ulrikstrid** [comment]
Fixed comments, formatted and rebased on main

File: flake.nix [hunk 4]
```diff
@@ -10,7 +10,10 @@
   inputs = {
     nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
     flake-utils.url = "github:numtide/flake-utils";
-    nix-npm-buildpackage.url = "github:serokell/nix-npm-buildpackage";
+    nix-filter.url = "github:numtide/nix-filter";
```
Comment (ulrikstrid) [inline flake.nix:13]: After we land this we should use `nix-filter` for the deku source to cut down on the rebuilds when changing unrelated files

**Review by d4hines — APPROVED** [review]
LGTM
